### ISSUE TITLE
Road markers in Organic Maps aren't always displayed in accordance with national practice

### ISSUE THREAD

**Issue by GCarty80** [organicmaps/organicmaps#1244]
For example, on UK maps all A-roads currently have their road markers displayed as amber on green in Organic Maps, even though on the ground only the more important A-roads (known as "primary routes") have their road markers displayed this way.

I suspect this may be because the OSM classification of "Primary Road" is actually used for _non-primary_ A-roads in the UK (with OSM "Secondary Road" used for B-roads). UK primary routes are classified within OSM as "Trunk Roads", so only this road categorization should get the amber-on-green road markers, with OSM "Primary Roads" getting the same black-on-white markers as "Secondary Roads".

How road markers are currently displayed -- "A688" is a primary route (implemented in OSM as a Trunk Road), "A177" is a non-primary A road (implemented in OSM as a Primary Road) and "B6291" is a B road (implemented in OSM as a Secondary Road):

![Screenshot_20210903-085131_Organic Maps](https://user-images.githubusercontent.com/90029834/132005218-2b5bee7e-b193-4188-868d-a33fb4c91d4b.jpg)
![Screenshot_20210903-085057_Organic Maps](https://user-images.githubusercontent.com/90029834/132005222-ef173f1a-1213-4c41-a983-35ec1ef7b323.jpg)

And here is how they _should_ be displayed:

![Screenshot_20210903-085131_Organic Maps Corrected](https://user-images.githubusercontent.com/90029834/132005238-cc6d97eb-3013-442d-a1ca-f475e3a2476e.jpg)
![Screenshot_20210903-085057_Organic Maps Corrected](https://user-images.githubusercontent.com/90029834/132005245-0963d894-bc42-46df-911f-527909020cd7.jpg)

**Comment by poeticus** [comment]
I completely agree with this post in terms of the labelling of A-roads in the UK, but would add that currently there are further issues specific to 'A-road motorways'.

Although these motorways should be labelled in blue, as in the screenshots above, they're currently showing in green as in the screenshot below of the same junction.

![Durham](https://user-images.githubusercontent.com/118109677/205507630-d47f74d2-d9db-46ba-8d2f-9ebc92a59c4e.png)

In addition, the A1 between London and Edinburgh is a single road with a number of either 'A1' or 'A1(M)' depending on whether or not it is a motorway section. The screenshot shows it incorrectly labelled with both road numbers.

This also affects other A-road motorways in the UK, perhaps most clearly on the motorway in SW Scotland that runs between the English border and Glasgow.

This is the 'A74(M)' for its southern half and the 'M74' for its northern half. The road number changes at the junction shown in the screenshot below. South of the junction it is not the M74 and so all the M74 labels south of here are incorrect and should not appear.

![Abington](https://user-images.githubusercontent.com/118109677/205507647-06e569df-b92f-4111-9187-372c4ecad176.png)

**Comment by pastk** [comment]
Actually in OM road markers colors don't depend on the road tag (primary/trunk/etc.).

It depend on marker's text and optionally on a road network specified in way/relation tags
(and there was a bug when "unknown" to OM road networks colored markers always white, regardless of text content, fixed in #6156).

ATM all UK markers that contain `A` are colored yellow/green, while all `M` are blue and the rest default to white.

But I understand that A roads in UK differ by number ranges also?

**Comment by pastk** [comment]
And what color the `A1 (M)` should be?
On your screenshot its blue, but actually since then it was changed to yellow-green by https://github.com/organicmaps/organicmaps/commit/43cf2383202

**Comment by poeticus** [comment]
There's no difference in signage for A roads in the UK related to number range. The difference is only between those officially designated as primary routes and those that are not. The 'UK tagging' section here sets it out nicely https://wiki.openstreetmap.org/wiki/Roads_in_the_United_Kingdom 

So A roads with primary status (signed yellow on green) are tagged as 'trunk', and those with non-primary status (signed black on white) are tagged as 'primary'.

Similarly, roads under motorway regulation (which includes those numbered in the form of both 'M1' and 'A1(M)') are tagged as 'motorway'. Signs for all of these should be blue.

Examples of UK signs illustrating this are shown here in the official Highway Code (need to scroll down quite a way).
https://www.gov.uk/guidance/the-highway-code/traffic-signs

**Comment by pastk** [comment]
> 
> So A roads with primary status (signed yellow on green) are tagged as 'trunk', and those with non-primary status (signed black on white) are tagged as 'primary'.

So there should be an official list somewhere telling which A roads has a primary/motorway/non-primary status?

**Comment by poeticus** [comment]
For some 'A-road motorways', like the A1(M), the further incorrect shield issue referenced above (#4028) sits alongside this one and is relevant. Here 2 road numbers are displayed at the same location (in this example both A1 and A1(M)). I assume that this may be due to the relation number (A1) also being displayed for the A1(M) sections.

**Comment by pastk** [comment]
Yes, one road ref is taken from the way's tag while the other comes from the relation and OM doesn't see them as the same (need a UK-specific parsing logic to normalize it to a single "standard" form).

**Comment by pastk** [comment]
@poeticus do you know or did you check if primary/motorway/non-primary status in UK is usually set in relations as a network tag somehow?

**Comment by poeticus** [comment]
I'm not an expert in the details of highways mapping, any knowledge I do have being what I've learned examining how highways have been tagged while trying to understand why some things are displayed the way they are.

With that in mind, the only place that I've seen highway status being defined is in a tag in each Way element, as highway=motorway (motorways), highway=trunk (Primary Routes), highway=primary (non-Primary Route A-roads), etc. 

The A40 relation covers the entire length of the A40 from central London to SW Wales and includes both Primary Route and non-Primary Route sections so, using this as an example, I'm not sure how the status could be set in the relation.

It should be noted that OM already distinguishes between Primary and non-Primary Routes by rendering the former in a brighter orange, even though the shields are currently the same.

Apologies if I've misunderstood your question here.

**Comment by poeticus** [comment]
A further note: there is a risk of course that some people may not like the suggested change to UK A-road shields as they're used to all A-roads being shown with a green shield. Having just checked, I can see this is still the case with Magic Earth and OsmAnd. (Magic Earth does show A-road motorways with a blue shield.) However, if the aim is for OM to match signage on the road then I remain in agreement with the original post that it does need to change. 

UK Ordnance Survey maps do make the distinction, for example here east of Oxford where the A40 changes its status.
https://explore.osmaps.com/?lat=51.738052&lon=-1.086759&zoom=12.1661&style=Standard&type=2d
